### DISCUSSION TITLE
APPLICATION FAILED TO START : SSL : Binding to target [Bindable@557a1e2d type = java.util.List<com.provectus.kafka.ui.config.ClustersProperties$Cluster>, value = 'provided', annotations = array[[empty]]] failed:

### DISCUSSION THREAD

**Discussion by thennati** [provectus/kafka-ui#2417] [Q&A] [answered]
Failed to collect cluster Default info error while creating Admin client for cluster Default

**Comment by Haarolean** [comment, marked as answer]
Please share full logs

**Reply by thennati** [reply]
WARN  [main] o.s.b.w.r.c.AnnotationConfigReactiveWebServerApplicationContext: Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'readOnlyModeFilter' defined in URL [jar:file:/kafka-ui-api.jar!/BOOT-INF/classes!/com/provectus/kafka/ui/config/ReadOnlyModeFilter.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'clustersStorage' defined in URL [jar:file:/kafka-ui-api.jar!/BOOT-INF/classes!/com/provectus/kafka/ui/service/ClustersStorage.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.boot.context.properties.ConfigurationPropertiesBindException: Error creating bean with name 'clustersProperties': Could not bind properties to 'ClustersProperties' : prefix=kafka, ignoreInvalidFields=false, ignoreUnknownFields=true; nested exception is org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'kafka.clusters' to java.util.List<com.provectus.kafka.ui.config.ClustersProperties$Cluster>
2022-08-09 07:25:27,499 INFO  [main] o.s.b.a.l.ConditionEvaluationReportLoggingListener:

 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2022-08-09 07:25:27,561 ERROR [main] o.s.b.d.LoggingFailureAnalysisReporter:

 

***************************
APPLICATION FAILED TO START
***************************

 

Description:

 

Binding to target [Bindable@557a1e2d type = java.util.List<com.provectus.kafka.ui.config.ClustersProperties$Cluster>, value = 'provided', annotations = array<Annotation>[[empty]]] failed:

 

   Property: kafka.clusters[0].bootstrapservers.ssl.trustore.location
    Value: /etc/pki/java/cacerts
    Origin: System Environment Property "KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS_SSL_TRUSTORE_LOCATION"
    Reason: The elements [kafka.clusters[0].bootstrapservers.ssl.trustore.location,kafka.clusters[0].bootstrapservers.ssl.trustore.password] were left unbound.
    Property: kafka.clusters[0].bootstrapservers.ssl.trustore.password
    Value: changeit
    Origin: System Environment Property "KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS_SSL_TRUSTORE_PASSWORD"
    Reason: The elements [kafka.clusters[0].bootstrapservers.ssl.trustore.location,kafka.clusters[0].bootstrapservers.ssl.trustore.password] were left unbound.

 

Action:

 

Update your application's configuration

**Reply by Haarolean** [reply]
it seems like you did set `KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS_SSL_TRUSTORE_LOCATION` but didn't set `KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS_SSL_TRUSTORE_PASSWORD`.

**Reply by michaelm-88** [reply]
@Haarolean need to be on the env variables ??
